kernel/packages/unity-interface/loader.ts

```diff
@@ -0,0 +1,105 @@
+import future from 'fp-future'
+import { parseUrn } from '@dcl/urn-resolver'
+
+declare const globalThis: any
+
+export type DclRenderer = {}
+export type UnityLoader = {}
+export type LoadRendererResult = {
+  DclRenderer: DclRenderer
+  UnityLoader: UnityLoader
+  buildConfigPath: string
+}
+
+/**
+ * This function is used by preview, instead of using "." as root,
+ * preview uses '/@/artifacts'
+ */
+function getRendererArtifactsRoot() {
+  if (typeof globalThis.RENDERER_ARTIFACTS_ROOT == 'undefined') {
+    throw new Error('RENDERER_ARTIFACTS_ROOT is undefined')
+  } else {
+    return new URL(globalThis.RENDERER_ARTIFACTS_ROOT, document.location.toString()).toString()
+  }
+}
+
+// TODO: return type DclRenderer
+async function injectRenderer(baseUrl: string): Promise<LoadRendererResult> {
+  const scriptUrl = new URL('index.js', baseUrl).toString()
+  const buildConfigPath = new URL('unity.json', baseUrl).toString()
+  await injectScript(scriptUrl)
+
+  if (typeof globalThis.UnityLoader == 'undefined') {
+    throw new Error('Error while loading UnityLoader from ' + scriptUrl)
+  }
+
+  if (typeof globalThis.DclRenderer == 'undefined') {
+    throw new Error('Error while loading the renderer from ' + scriptUrl)
+  }
+
+  return {
+    DclRenderer: globalThis.DclRenderer,
+    UnityLoader: globalThis.UnityLoader,
+    buildConfigPath
+  }
+}
+// TODO: return type DclRenderer
+async function loadDefaultRenderer(): Promise<LoadRendererResult> {
+  // TODO: after we remove unity from this folder, remove the following
+  // block and uncomment the block next to it
+  {
+    const scriptUrl = new URL('DCLUnityLoader.js', getRendererArtifactsRoot()).toString()
+    const buildConfigPath = new URL('unity.json', getRendererArtifactsRoot()).toString()
+    await injectScript(scriptUrl)
+    if (typeof globalThis.UnityLoader == 'undefined') {
+      throw new Error('Error while loading the renderer from ' + scriptUrl)
+    }
+    return {
+      DclRenderer: globalThis.DclRenderer,
+      UnityLoader: globalThis.UnityLoader,
+      buildConfigPath
+    }
+  }
+
+  // {
+  //   // Load the renderer from the artifacts root folder
+  //   return await injectRenderer(getRendererArtifactsRoot())
+  // }
```

Comment:
```suggestion
```

Reply:
This should remain, read the TODO a couple lines above